utils.py

```diff
@@ -2111,61 +2111,67 @@ def minimal_realisation(a, b, c):
 
     return Aco, Bco, Cco
 
-
 def num_denom(A, symbolic_expr=False):
+    """
+    Returns the numerator and denominator polynomial
+    of a utils.tf or utils.mimotf object. Can also return a simple
+    symbolic expression for the numerator and denominator.
+
+    :param A: utils.tf or utils.mimotf object
+    :param symbolic_expr: Boolean, false by default. If true returns a
+        symbolic representation.
+    :return: First returned value is the numerator, second is the
+        denominator
+    """
 
     sym_den = 0
     sym_num = 0
     s = sympy.Symbol('s')
 
     if type(A) == mimotf:
-        denom = 1
-        num = 1
-
-        denom = [numpy.poly1d(denom) *
-                 numpy.poly1d(A.matrix[0, j].denominator.coeffs)
-                 for j in range(A.matrix.shape[1])]
-        num = [numpy.poly1d(num) *
-               numpy.poly1d(A.matrix[0, j].numerator.coeffs)
-               for j in range(A.matrix.shape[1])]
-        if symbolic_expr is True:
-            for n in range(len(denom)):
-                sym_den = (sym_den + denom[- n - 1] * s**n).simplify()
-            for n in range(len(num)):
-                sym_num = (sym_num + num[- n - 1] * s**n).simplify()
-            return sym_num, sym_den
-        else:
-            return num, denom
-
+        # Ugly hack to get mimo size until class is rewritten...
+        rows, cols = 0, 0
```

Comment:
What's wrong with `rows, cols = A.matrix.shape`

Reply:
Sorry, I assumed that function would return incorrectly because of the mimotf class' strange behaviour when being indexed:

`g1 = tf([1],[1,2])`
`g2 = tf([2],[1,3])`
`g3 = tf([3],[1,4])`
`g4 = tf([4],[1,5])`
`g5 = tf([5],[1,6])`
`g6 = tf([6],[1,7])`

`G = mimotf([`
`    [g1, g2],`
`    [g3, g4],`
`    [g5, g6]`
`])`

`>>>G`
`[1] : mimotf([[tf([ 1.], [ 1.  2.]) tf([ 2.], [ 1.  3.])]`
 `[tf([ 3.], [ 1.  4.]) tf([ 4.], [ 1.  5.])]`
 `[tf([ 5.], [ 1.  6.]) tf([ 6.], [ 1.  7.])]])`

`>>>G[0]`
`[2] : mimotf([[tf([ 1.], [ 1.  2.]) tf([ 2.], [ 1.  3.])]])`

`>>>G[0][0]`
`[3] : mimotf([[tf([ 1.], [ 1.  2.]) tf([ 2.], [ 1.  3.])]`

`>>>G[0][0][0][0][0][0][0][0]`
`[4] : mimotf([[tf([ 1.], [ 1.  2.]) tf([ 2.], [ 1.  3.])]])`

After seeing this behaviour and struggling for a while to have things behave correctly I found the the mimotf_slice() function worked as expected and thought no further. Sorry, I should have tried the .shape property but I forgot the mimotf class was derived from a matrix after seeing the indexing behaviour.